### ISSUE TITLE
[Resource] sorting via routing doesn't work with pagination and limit disabled

### ISSUE THREAD

**Issue by peteward** [Sylius/Sylius#4072]
This doesn't sort:

```
reiss_backend_store_list:
    path: /
    methods: [GET]
    defaults:
        _controller: reiss.controller.store:indexAction
        _sylius:
            template: ReissStoreLocatorBundle:Backend/Store:index.html.twig
            paginate: false
            limit: false
            sorting:
                name: asc
```

A workaround is to set an arbitrarily high limit:

```
reiss_backend_store_list:
    path: /
    methods: [GET]
    defaults:
        _controller: reiss.controller.store:indexAction
        _sylius:
            template: ReissStoreLocatorBundle:Backend/Store:index.html.twig
            paginate: false
            limit: 2000
            sorting:
                name: asc
```

**Comment by gperdomor** [comment]
Similar issue... I have in my routes something like this:
<pre><code>customer_index:
    path: /
    methods: [GET]
    defaults:
        _controller: sylius.controller.customer:indexAction
        _sylius:
            template: ...
            sortable: true
            sorting:
                firstName: asc
</code></pre>

And the collection is sort properly, but in the view, the sort icon is in <code>id</code> column and not in <code>firstName</code> column.

This only affect sylius 0.17

**Comment by pamil** [comment]
Solved by #7168.
